ui/src/login/login.tsx

```diff
@@ -1,20 +1,55 @@
 import {Page} from 'argo-ui/src/components/page/page';
 import * as React from 'react';
+import {useState} from 'react';
 
 import {uiUrl, uiUrlWithParams} from '../shared/base';
+import {services} from '../shared/services';
 import {useCollectEvent} from '../shared/use-collect-event';
 
 import './login.scss';
 
+// Clear any cached data on logout
 function logout() {
     document.cookie = 'authorization=;Max-Age=0';
-    document.location.reload();
+    // Clear any cached API data
+    services.info.clearCache();
+    // Add a small delay before reload to ensure cookies are cleared
```

Comment:
What cookies do you mean by this, and why would we need a delay before they're cleared?